app/forms/qualification_details_form.rb

```diff
@@ -0,0 +1,143 @@
+class QualificationDetailsForm < Form
+  attribute :qualifications_details_check, :boolean
+
+  validates :qualifications_details_check,
+    inclusion: {
+      in: [true, false],
+      message: ->(form, _) { form.i18n_errors_path("qualifications_details_check") }
+    }
+
+  def initialize(claim:, journey:, params:)
+    super
+
+    self.qualifications_details_check = permitted_params.fetch(
+      :qualifications_details_check,
+      claim.qualifications_details_check
+    )
+  end
+
+  def save
+    return false unless valid?
+
+    claim.assign_attributes(
+      qualifications_details_check: qualifications_details_check
+    )
+
+    if qualifications_details_check
+      # Teacher has confirmed the details in the dqt record are correct, update
+      # the eligibility with these details
+      claim.claims.each { |c| set_qualifications_from_dqt_record(c.eligibility) }
+    else
+      # Teacher has said the details don't match what they expected so
+      # nullify them
+      claim.claims.each { |c| set_nil_qualifications(c.eligibility) }
+    end
+
+    claim.save!
```

Comment:
I think we should try to use the inherited `update!` method as much as possible, as it's going to make it easier to migrate the persistence layer for the claim (`CurrentClaim`) and minimise changes around. It might not be easy or doable right now, so just checking

Reply:
I'm not sure if we can as on the additional payments journey we set different attributes on the ECP and LUP, with the main issue being the `eligible_degree_subject` attribute. Setting `eligible_degree_subject` requires calling `Policies::LevellingUpPremiumPayments::DqtRecord#eligible_degree_code?` which doesn't have a corresponding method on the ECP DqtRecord. Will see if there's some other way to reuse the update method though